frontend/main.js

```diff
@@ -201,12 +201,16 @@ async function startApp () {
     sbp('okTurtles.data/set', PUBSUB_INSTANCE, sbp('chelonia/connect', {
       messageHandlers: {
         [NOTIFICATION_TYPE.VERSION_INFO] (msg) {
+          const isProduction = process.env.NODE_ENV === 'production'
           const ourVersion = process.env.GI_VERSION
           const theirVersion = msg.data.GI_VERSION
 
           const ourContractsVersion = process.env.CONTRACTS_VERSION
           const theirContractsVersion = msg.data.CONTRACTS_VERSION
-          if (ourVersion !== theirVersion || ourContractsVersion !== theirContractsVersion) {
+
+          const isContractVersionDiff = ourContractsVersion !== theirContractsVersion
+          const isGIVersionDiff = ourVersion !== theirVersion
+          if (isContractVersionDiff || (!isProduction && isGIVersionDiff)) {
```

Comment:
Why is the production environment a factor here?

Reply:
Agreed, I have the same question

Reply:
I am not sure why we need `GI_VERSION`, but we are currently using it for version management. As we can see the definition of `GI_VERSION` below (it's in Gruntfile.js), it uses the current date.
```
const GI_VERSION = packageJSON.version + (NODE_ENV === 'production' ? `@${new Date().toISOString()}` : '')
```
So the `GI_VERSION` would be different each time of calling `grunt deploy`, `grunt serve`. Because of the difference in two `GI_VERSION`s, the `NOTIFICATION_TYPE.VERSION_INFO` event is called, and the page is refreshing infinitely.
```
sbp('okTurtles.events/on', NOTIFICATION_TYPE.VERSION_INFO, (versionInfo) => {
  console.info('New Group Income version available:', versionInfo)
  // Prevent the client from trying to reconnect when the page starts unloading.
  sbp('okTurtles.data/get', PUBSUB_INSTANCE).destroy()
  // TODO: allow the user to manually reload the page later.
  window.location.reload()
})
```

Reply:
I see. Could you please then check, instead of `'production'` for `'development'` (e.g. `process.env.NODE_ENV === 'development'`)?

And then update the `if` condition plus add an explanatory comment:

```js
// We only compare GI_VERSION in development mode so that the page auto-refreshes if `grunt dev` is re-run
// This check cannot be done in production mode as it would lead to an infinite page refresh bug
// when using `grunt deploy` with `grunt serve`
if (isContractVersionDiff || (isDevelopment && isGIVersionDiff)) {
```

Reply:
Sure.